src/TinyIoC/TinyIoC.cs

```diff
@@ -3905,87 +3944,97 @@ private object ResolveInternal(TypeRegistration registration, NamedParameterOver
         }
 
 #if EXPRESSIONS
-        private object GetLazyAutomaticFactoryRequest(Type type)
-        {
-            if (!type.IsGenericType())
-                return null;
+    private readonly SafeDictionary<Type, object> _LazyAutomaticFactories = new SafeDictionary<Type, object>();

```

Comment:
This looks like it could grow exponentially, I wonder if we should do anything to allow these to be evicted overtime or limit it to cache the last few hundred? What are the thoughts on this?

Reply:
> This looks like it could grow exponentially
I'm not sure about exponentially, but you're right there's no limiting factor. I don't think there is on the existing object constructor cache either (unless I missed it). I'm not sure how many factories consumers are likely to request (I assumed fewer than the total number of registered types, you don't always need a factory and it should only add when a factory is requested?). However I wouldn't be at all opposed to some kind of eviction. It would potentially make the performance less *predictable* if an eviction happened and then the type was requested again,  but I think it would still be a win overall. I would suggest perhaps a LRU style eviction with some max cap rather than a timed based eviction, but that's just my 2cents. 

Depending on usage, I think this would normally have fewer items in it that the existing object constructor cache so I'm not sure if an eviction process is needed or not? 

> I guess it also doesn't matter if it's thread safe as more work will be done and it will be eventually consistent
Yes, that was my thinking, and I believe the existing/similar cache for the expression based object constructors works on the same idea. I seem to recall there was even a comment to the effect it was assumed it wasn't worth locking on each request for the occasional/potential double creation if a race actually happened.

Reply:
Also, you're welcome, and thanks for taking the time to review and consider the changes :)